products/sle/main.pm

```diff
@@ -292,11 +292,11 @@ if (sle_version_at_least('15') && !check_var('SCC_REGISTER', 'installation')) {
             my $module_repo_name = get_var($repo_variable_name, $default_repo_name);
             my $url = "$utils::OPENQA_FTP_URL/$module_repo_name";
             # Verify if url exists before adding
-            if (head($url)) {
-                set_var('ADDONURL_' . uc $short_name, "$utils::OPENQA_FTP_URL/$module_repo_name");
-                $addonurl .= "$short_name,";
-            }
+            die "URL $url could not be accessed, missing repo?" unless head($url);
+            set_var('ADDONURL_' . uc $short_name, "$utils::OPENQA_FTP_URL/$module_repo_name");
+            $addonurl .= "$short_name,";
         }
+        die '$addonurl could not be set' unless $addonurl;
```

Comment:
Let's put variable name in the message too. Otherwise LGTM.

Reply:
That's what I'm saying. No one will get what is $addonurl without looking in the code, but ADDONURL would be more clear imho.

Reply:
ADDONURL is also empty. Do you mean I should write

```perl
die '$addonurl (test variable \'ADDONURL\') could not be set' unless $addonurl;
```

?

Reply:
Even though I don't know what @rwx788 exactly means: I like your suggestion @okurz - could make it easier to understand what happened.